lib/calculate-cache-key-for-tree.js

```diff
@@ -16,10 +16,28 @@ const stringify = require('json-stable-stringify');
   @return {String} cacheKey
 */
 module.exports = function calculateCacheKeyForTree(treeType, addonInstance, additionalCacheKeyParts) {
+  let pkg = addonInstance.pkg || {};
+
+  let {
+    name,
+    version,
+    dependencies,
+    devDependencies,
+    peerDependencies,
+    bundledDependencies,
+    optionalDependencies
+  } = pkg;
+
   let cacheKeyParts = [
-    addonInstance.pkg,
     addonInstance.name,
-    treeType
+    treeType,
+    name,
```

Comment:
I am not sure if `name` is entirely redundant with `addonInstance.name`, so I cautiously left both.

Reply:
Your caution has paid off 😸, they can indeed be different (though we are pushing very hard to ensure that this is "going away").